include/realizations/catchment/Bmi_Fortran_Adapter.hpp

```diff
@@ -675,7 +675,7 @@ namespace models {
 
                 // Must get the names from the model as an array of C strings
                 // The array can be on the stack ...
-                char* names_array[variableCount];
+                std::vector<char*> names_array(variableCount);
                 // ... but allocate the space for the individual C strings (i.e., the char * elements)
                 for (int i = 0; i < variableCount; i++) {
                     names_array[i] = static_cast<char *>(malloc(sizeof(char) * BMI_MAX_VAR_NAME));
```

Comment:
This is a hpp file so Is there a reason this is not using `new char[BMI_MAX_VAR_NAME]` or
`typedef bmi_var_name_str char[BMI_MAX_VAR_NAME]`
`std::vector<bmi_var_name_str> names_array`

Reply:
There is no need to use c memory functions particularly when the allocation size for a name is fixed